mangaki/mangaki/models.py

```diff
@@ -89,6 +89,20 @@ def random(self):
             nb_dislikes__lte=RANDOM_MAX_DISLIKES,
             nb_likes__gte=F('nb_dislikes') * RANDOM_RATIO)
 
+    def group_by_category(self):
+        """
+        Groups this queryset by category.
+
+        Returns:
+          by_category -- A mapping from category IDs to the list of works in
+              this queryset in the corresponding category. Order inside a
+              category is preserved.
+        """
+        by_category = {}
```

Comment:
It's not going to be possible to return a queryset with the right structure (mapping from categories to list of works) anyways. I'll precise in the docstring that the mapping this returns is a dict.